### DISCUSSION TITLE
PVE下IKUAI+adguardhome+paopaodns+ppgw配置分享

### DISCUSSION THREAD

**Discussion by sunshinehcyyy** [kkkgo/PaoPaoDNS#158] [经验分享]
PS：不会使用编辑器，大家凑合着看。
1. 配置说明

PVE 10.0.0.254
IKUAI 10.0.0.1 （PVE虚拟机）
AdGuardHome 10.0.1.2（基于ikuai内置的docker搭建）
ppdns 10.0.0.5（基于pve虚拟机，这里我本来也想用ikuai自带的虚拟机，但是一直设置都不成功）
ppgw 10.0.0.2PVE虚拟机

2. ikuai初步设置

首先PVE新建一个Ikuai虚拟机，设置好能上网（已经是ikuai主路由的忽略），然后去ikuai云装上docker，并且下载adguardhome的镜像，这里注意，不要下载默认的latest版本，手动选择最新版v0.107.50。不知道是我用了阿里云的镜像加速还是什么原因，直接下载latest版本，显示是很早的版本了，介意的就手动选版本，不介意的就直接latest。然后就是跑adguardhome，这个没啥写的，自己设置好黑名单，然后adguardhome的DNS上游先设置成公共dns。
再在ikuai的dns里面把DHCP服务器里面的DNS都指向10.0.0.1（也就是你们自己的ikuai地址），ikuai自己的DNS设置全写adguardhome 的地址，我这般是10.0.1.2.这里我看大部分的教程都是后期指向PPDNS，但是我实测，都写ikuai地址，然后ikuai通过adguardhome解析地址，效果是一样的。
![DHCP](https://github.com/kkkgo/PaoPaoDNS/assets/50049108/e997d7d1-3b1d-495c-9163-0e941145c367)
![IKUAI DNS](https://github.com/kkkgo/PaoPaoDNS/assets/50049108/d6cfc253-a08a-4ded-a9dc-fd6dd7164ea9)

3. 搭建ppdns

我这里用的是这个的教程。
(https://www.jackiewu.top/45fe9747966b48f5a4b9ec08fb1fdd80?pvs=25#d5a9bd3531b540a4ba5b76eed05c39b3)
设置完之后执行
```
docker exec paopaodns test.sh
```

如果显示
```
ALL TEST PASS
```
就可以把adguardhome 的上游dns 指向ppdns的IP了。
tips：我这里还在这个虚拟机搭建了Portainer，需要的也可以自行搭建。便于管理。

4.创建ppgw并启动
这里我直接用群晖的docker创建的，
network.ini配置
```
ip=10.0.0.2
mask=255.255.255.0
gw=10.0.0.1
dns1=10.0.0.1
```
ppgwurl.ini配置
```
ppgwurl="http://XXXXXXX/ppgw.ini"
```
这里我让ppgw启动之后直接到我文件服务器取配置文件，方便后期我对配置进行修改。
ppgw.ini配置
```
#paopao-gateway
mode="suburl"" #这里是说采取订阅模式获取节点信息
fake_cidr="11.0.0.0/8"
dns_ip="10.0.0.5" #这里写ppdns的地址
dns_port="5304"
clash_web_port="80"
clash_web_password="123456"
openport="yes"
udp_enable="yes"

# default:30
sleeptime="30"

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="10.0.0.2"
# default: socks5_port="7890"
socks5_port="7890"
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

yamlfile="custom.yaml"

# suburl mode settting
suburl="http://10.0.0.1/conf.yaml"#这里节点订阅地址
subtime="1d"

# fast_node=check/yes/no
fast_node=yes
test_node_url="https://www.youtube.com/generate_204"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"

# dns burn setting
# depand on fast_node=yes & mode=suburl/yaml
dns_burn=yes
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="10.0.0.5:53" #这里写ppdns的地址
net_rec="yes"
max_rec="5000"
```

生成ISO之后就可以直接扔PVE启动了，只需要基础的虚拟机配置，ISO和一个网口，不需要硬盘，这里默认应该是没有任何问题的。

5. 设置Ikuai静态路由
这里我没有像大家一样，设置了11.0.0.0/8和telegram+Netflix到静态路由，我只设置了10.0.0.0/8的静态路由，telegram和Netflix我是通过端口分流的。
![静态路由](https://github.com/kkkgo/PaoPaoDNS/assets/50049108/f0c5f5ca-368b-47c8-964f-b14eb032311a)

![NAT](https://github.com/kkkgo/PaoPaoDNS/assets/50049108/c0d754b0-725a-41f8-98ed-e106e6b8c600)
![端口分流](https://github.com/kkkgo/PaoPaoDNS/assets/50049108/8607b83a-4db6-483a-a7be-e935d5037725)

至此完结。

**Comment by runishiwo** [comment]
PPDNS前面放了ADGHOME，DHCP首选DNS设置到10.0.0.1 ikuai的地址后，ADGHOME查询日志里面没有任何新增的DNS请求，感觉还是得DHCP首选DNS直接设置为ADGHOME的IP地址
